src/lib/array.jl

```diff
@@ -100,21 +100,21 @@ end
 Base.getindex(xs::TrackedArray, i...; kwargs...) = track(getindex, xs, i...; kwargs...)
 
 @grad function getindex(xs::AbstractArray, i...; kwargs...)
-  data(xs)[i...; kwargs...], function (Δ)
+  getindex(data(xs), i...; kwargs...), function (Δ)
     Δ′ = zero(xs)
-    Δ′[i...] = data(Δ)
-    (nobacksies(:getindex, Δ′), map(_->nothing, i)...)
-  end
+    setindex!(Δ′, data(Δ), i...; kwargs...)
+    (nobacksies(:getindex, Δ′), map(_->nothing, i)...)  # TODO: put kwargs here
```

Comment:
The todo is unecessary, you don't need to provide gradients for kwargs.

Reply:
why not?

Reply:
We just assume that they are non-differentiable; it's inconvenient to provide them and easier to just create a non-kwargs method if you need that.

Do you need differentiable kwargs here?

Reply:
Not for any reasonable definition of indexing